lib/chef/knife/vsphere_vm_snapshot.rb

```diff
@@ -67,14 +73,22 @@ def run
     end
 
     if config[:create_new_snapshot]
-      vm.CreateSnapshot_Task(name: config[:create_new_snapshot], description: '', memory: false, quiesce: false)
+	if config[:wait]
+      vm.CreateSnapshot_Task(name: config[:create_new_snapshot], description: '', memory: false, quiesce: false).wait_for_completion
```

Comment:
I haven't tried it, but instead of doing

```
if option
  task.wait_for_completion
else
  task
end
```

is it possible to reduce the redundancy with:
```
snapshot = task
snapshot.wait_for_completion if option
```

There's another open PR that works with some of the options (`memory` and `quiesce`) and I'd like to avoid an explosion of options, if at all possible.